main.py

```diff
@@ -141,19 +323,32 @@ def heartbeat() -> None:
         # The lru_cache sometimes uses a lot of memory, making normal exit slow
         # due to GC. But since we're sane people and don't use finalizers for
         # cleanup, we can just skip GC and exit immediately.
-        os._exit(0)
+        #os._exit(0)
+
+iteration = 0
+errors = 0
+overall_profiler = Profiler()
+high_scores = {}
+permuters: List[Permuter] = []
```

Comment:
global state :( can we avoid it?

Reply:
Someone smarter probably can, but I ran into a lot of issues passing context to a new thread as opposed to it being forked :(